cli/tests/test_mcp_dns_pinning.py

```
@@ -0,0 +1,178 @@
+        monkeypatch.delenv(name, raising=False)
+
+    with _serve_mcp(transport) as port:
+        real_getaddrinfo = socket.getaddrinfo
+        target_calls: list[str] = []
+        pinned_calls: list[str] = []
+        analyzer_calls: list[str] = []
+        unexpected_calls: list[str] = []
+        analyzers_started = 0
+        both_analyzers_started = asyncio.Event()
+
+        def rebinding_resolver(host, service, *args, **kwargs):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `rebinding_resolver`

(ANN202)